### PR TITLE
feat(DEW): csms secret resource support secret_binary and expire_time parameter

### PR DESCRIPTION
**Pull request by MrLiangSir**
**What this PR does / why we need it**:
csms secret resource support secret_binary and expire_time parameter
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Exactly one of `secret_text` and `secret_binary` should be specified.
2. The expire_time can be update only when the secret key state is ENABLED.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dew TESTARGS='-run TestAccDewCsmsSecret_SecretBinaryAndExpireTime'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_SecretBinaryAndExpireTime -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_SecretBinaryAndExpireTime
=== PAUSE TestAccDewCsmsSecret_SecretBinaryAndExpireTime
=== CONT  TestAccDewCsmsSecret_SecretBinaryAndExpireTime
--- PASS: TestAccDewCsmsSecret_SecretBinaryAndExpireTime (129.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       129.402s

 make testacc TEST=./huaweicloud/services/acceptance/dew TESTARGS='-run TestAccDewCsmsSecret_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_basic -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_basic
=== PAUSE TestAccDewCsmsSecret_basic
=== CONT  TestAccDewCsmsSecret_basic
--- PASS: TestAccDewCsmsSecret_basic (61.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       61.565s
```

### CODE REVIEW

File: docs/resources/csms_secret.md [hunk 1]
```diff
@@ -29,6 +29,15 @@ resource "huaweicloud_csms_secret" "test2" {
 }
 ```
 
+### Encrypt String Binary
+
+```hcl
+resource "huaweicloud_csms_secret" "test3" {
+  name          = "test_binary"
+  secret_binary = "1010"
```
Comment (deer-hang) [inline docs/resources/csms_secret.md]: Please using variable value.

File: docs/resources/csms_secret.md [hunk 2]
```diff
@@ -39,9 +48,17 @@ The following arguments are supported:
...
   Only digits, letters, underscores(_), hyphens(-) and dots(.) are allowed.
 
-* `secret_text` - (Required, String) The plaintext of a secret in text format. The maximum size is 32 KB.
+* `secret_text` - (Optional, String) The plaintext of a secret in text format. The maximum size is 32 KB.
+
+* `secret_binary` - (Optional, String) The plaintext of a binary secret in text format encoded using Base64.
+  The maximum size is 32 KB.
+
+-> **NOTE:** Exactly one of `secret_text` and `secret_binary` should be specified.
+  The `secret_text` and `secret_binary` are sensitive and in the state file we store its hash.
```
Comment (deer-hang) [inline docs/resources/csms_secret.md:57]: remove this line.

File: docs/resources/csms_secret.md [hunk 3]
```diff
@@ -39,9 +48,17 @@ The following arguments are supported:
...
 
-  -> **NOTE:** The `secret_text` is sensitive and in the state file we store its hash.
+* `expire_time` - (Optional, Int) Specifies the expiration time of a secret. The time is in
+  the format of timestamp, that is, the offset milliseconds from 1970-01-01 00:00:00 UTC to the specified time.
```
Comment (deer-hang) [inline docs/resources/csms_secret.md]: Please confirm whether the unit is seconds or milliseconds.

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 4]
```diff
@@ -257,16 +292,44 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
...
-	if d.HasChanges("secret_text") {
+	if d.HasChanges("secret_text", "secret_binary") {
 		opts := secrets.CreateVersionOpts{
-			SecretString: d.Get("secret_text").(string),
+			ExpireTime: d.Get("expire_time").(int),
+		}
+		if v, ok := d.GetOk("secret_text"); ok {
+			opts.SecretString = v.(string)
+		} else {
+			opts.SecretBinary = d.Get("secret_binary").(string)
```
Comment (deer-hang) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go]: There is no need to add judgment, just splice it directly into the struct.

Same to create body

File: docs/resources/csms_secret.md (hunk 3 above)
Comment (deer-hang) [reply docs/resources/csms_secret.md]: expire_time can only be edited when `status` is **ENABLED**.

File: huaweicloud/services/dew/resource_huaweicloud_csms_secret.go [hunk 5]
```diff
@@ -257,16 +292,44 @@ func resourceCsmsSecretUpdate(ctx context.Context, d *schema.ResourceData, meta
...
+		// Sort by created time in descending order.
+		sort.Slice(versions, func(i, j int) bool {
+			return versions[i].CreateTime > versions[j].CreateTime
+		})
+
+		versionID := versions[0].ID
+		_, err = secrets.UpdateSecretVersion(client, name, versionID, opts)
+		if err != nil {
+			return diag.Errorf("failed to update the expire time of CSMS secret: %s", err)
+		}
```
Comment (deer-hang) [inline huaweicloud/services/dew/resource_huaweicloud_csms_secret.go:310]: 1、Please write a new method.
2、Please preCheck whether versions is empty.

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dew TESTARGS='-run TestAccDewCsmsSecret_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_ -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_basic
=== PAUSE TestAccDewCsmsSecret_basic
=== RUN   TestAccDewCsmsSecret_SecretBinaryAndExpireTime
=== PAUSE TestAccDewCsmsSecret_SecretBinaryAndExpireTime
=== CONT  TestAccDewCsmsSecret_basic
=== CONT  TestAccDewCsmsSecret_SecretBinaryAndExpireTime
--- PASS: TestAccDewCsmsSecret_basic (89.09s)
--- PASS: TestAccDewCsmsSecret_SecretBinaryAndExpireTime (117.59s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       117.646s

**Comment by MrLiangSir** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dew TESTARGS='-run TestAccDewCsmsSecret_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dew -v -run TestAccDewCsmsSecret_ -timeout 360m -parallel 4
=== RUN   TestAccDewCsmsSecret_basic
=== PAUSE TestAccDewCsmsSecret_basic
=== RUN   TestAccDewCsmsSecret_SecretBinaryAndExpireTime
=== PAUSE TestAccDewCsmsSecret_SecretBinaryAndExpireTime
=== CONT  TestAccDewCsmsSecret_basic
=== CONT  TestAccDewCsmsSecret_SecretBinaryAndExpireTime
--- PASS: TestAccDewCsmsSecret_basic (106.67s)
--- PASS: TestAccDewCsmsSecret_SecretBinaryAndExpireTime (138.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dew       138.924s

File: docs/resources/csms_secret.md [hunk 6]
```diff
@@ -39,9 +50,17 @@ The following arguments are supported:
 * `name` - (Required, String, ForceNew) The secret name. The maximum length is 64 characters.
   Only digits, letters, underscores(_), hyphens(-) and dots(.) are allowed.
 
-* `secret_text` - (Required, String) The plaintext of a secret in text format. The maximum size is 32 KB.
+* `secret_text` - (Optional, String) The plaintext of a secret in text format. The maximum size is 32 KB.
+
+* `secret_binary` - (Optional, String) The plaintext of a binary secret in text format encoded using Base64.
```
Comment (jinyangyang222) [inline docs/resources/csms_secret.md:55]: Parameter description starts with Specifies

**Pull request merged**